### DISCUSSION TITLE
Plotting the Schmid factor for twinning on the IPF during compression test！

### DISCUSSION THREAD

**Discussion by Jiangwenxiang123** [mtex-toolbox/mtex#1490] [Ask Anything]
Is there any way to plot the Schmid factor for twinning contour on the existing IPF sector during the compression test?

Thank you very much! 

Following is the code, however it is not correct! 
**how do I modify it? or could someone help me.**

cs = crystalSymmetry('cubic',[3.523,3.523,3.523],'mineral','Nickel')
% twin plane
n=Miller({1,1,1},{1,1,1},{1,1,1},{-1,1,1},{-1,1,1},{-1,1,1},{1,-1,1},{1,-1,1},{1,-1,1},{1,1,-1},{1,1,-1},{1,1,-1},cs);

% twin direction
b=Miller({1,1,-2},{1,-2,1},{-2,1,1},{-1,1,-2},{-1,-2,1},{2,1,1},{-2,-1,1},{1,2,1},{1,-1,-2},{-2,1,-1},{1,-2,-1},{1,1,2},cs);

%use function of slip
sS = slipSystem(n,b)

r = plotS2Grid(fundamentalSector(cs),'resolution',0.2*degree)

% compute the Schmid factors for all slip systems and all tension directions
tau = SchmidFactor(sS,r);

% tau is a matrix with columns representing the Schmid factors for the
% different twin systems. Lets take the maximum rhowise
[tauMax,id] = max(abs(tau),[],2);
contourf(r,tauMax,'xAxisDirection','south','contourf',0.2:0.008:0.5,'grid','grid_res',0*degree,'ShowText','on','LineColor','black','Fill','on');hold all
mtexColorbar

**Obtained by the above code** not correct!
![image](https://user-images.githubusercontent.com/90229941/179380877-e99e3f1f-df6b-4cb8-ab69-ddaf08abac2d.png)

**reported in reference** correct!
![image](https://user-images.githubusercontent.com/90229941/179380907-c54c6616-f2cd-4b6a-a767-84006d8ad368.png)

**Comment by zmichels** [comment]
at the link below, there is a method for Schmid Factor that uses a stress tensor (there are more than one methods/approaches detailed on that page). I would maybe start with that approach, but modify the stress tensor to match a case of compression.

https://mtex-toolbox.github.io/SchmidtFactor.html
